### DISCUSSION TITLE
Synced audio by means of special attribute

### DISCUSSION THREAD

**Discussion by vokimon** [asciinema/discussions#7] [Show and tell]
I wanted to synchronize audio with asciinema-player to narrate what is shown in the cast. In asciinema/asciinema-player#12, some hooks are proposed to get this working. I couldn't find any working example and I struggled to have them aside. When dealing with several casts, javascript could get messy relating audio tags and asciinema-player by id's. So, using the hooks worked but it was so complex.

To simplify that, I ended up with a [solution](https://github.com/vokimon/asciinema-player-audio) (sadly in javascript, not clojurejs) which locates any `asciinema-player` tag having the new `audiosrc` attribute, creates a hidden `audio` tag for each one and synchronizes their playback.

```html
<asciinema-player
	src="mycast.cast"
	audiosrc="mycast.mp3"
	audiotype="audio/mpeg"
/>
```
The sync code may require improvements far from my expertise, but yet it works for me, and for anyone reusing the script along with asciinema-player.

Now i wonder: being just few lines of JS code and given the convenience of syncing audio that easy, could it be a good idea to have it working in the main project?

Thanks, and congratulations for creating such an amazing tool.

**Comment by ku1ik** [comment]
I agree native support for audio would be handy :+1: 

Meanwhile there's a snippet integrating new v3 player with `<audio>` tag in asciinema/asciinema-player#159, should do the trick for now.

**Comment by ku1ik** [comment]
FYI asciinema player supports synced audio since version 3.11.

https://github.com/asciinema/asciinema-player/releases/tag/v3.11.0

You can use it on asciinema.org too - in recording settings there's "Audio URL" option.
